File: C# Fundamentals/C# OOP Advanced/07.Iterators_Comparators/Ex/P08_PetClinic/Pet.cs
using System;

public class Pet : IComparable<Pet>
{
    public string Name { get; set; }

    public int Age { get; set; }

    public string Kind { get; set; }

    public Pet(string name, int age, string kind)
    {
        this.Name = name;
        this.Age = age;
        this.Kind = kind;
    }

    public override string ToString()
    {
        return $"{this.Name} {this.Age} {this.Kind}";
    }

    public int CompareTo(Pet other)
    {
        int result = this.Name.CompareTo(other.Name);
        if(result == 0)
        {
            result = this.Age.CompareTo(other.Age);
        }
        return result;
    }
}

File: Java/01.Advanced/01.Advanced/17.IteratorsComparators/Ex/src/t_08_pet_clinics/Pet.java
package t_08_pet_clinics;

public class Pet implements Comparable<Pet>{
    private final String name;

    private final int age;

    private final String kind;

    public Pet(String name, int age, String kind)
    {
        this.name = name;
        this.age = age;
        this.kind = kind;
    }

    @Override
    public String toString()
    {
        return String.format("%s %d %s", name, age, kind);
    }

    public int compareTo(Pet other) {
        int result = this.name.compareTo(other.name);

        if (result == 0)
        {
            return Integer.compare(this.age, other.age);
        }

        return result;
    }
}
